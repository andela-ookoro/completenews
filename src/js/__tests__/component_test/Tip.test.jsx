import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import Tip from '../../pages/headlines/WelcomeTip';

describe('rendering', () => {
  it('should render content as describe in the component', () => {
    const component = Renderer.create(
      <Tip />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  const component = shallow(<Tip />);
  it('should render a welcome text', () => {
    const WelcomeText = component.find('.tipHeading').text();
    expect(WelcomeText).toBe(' Welcome to Complete news');
  });

  const tipText = component.find('p');

  it('should display a text to login', () => {
    const loginTip = tipText.nodes[2].props.children;
    expect(loginTip).toBe('Login to store interesting articles to your dashboard');
  });

  it('should display an arrow pointing towards the sources panel', () => {
    const arrowToSource = tipText.nodes[5].props.children;
    expect(arrowToSource).toBe(' ‚Üê ');
  });
});
