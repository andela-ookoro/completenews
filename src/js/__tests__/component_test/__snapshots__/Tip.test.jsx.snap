// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render content as describe in the component 1`] = `
<div
  className="tip"
>
  <div
    className="tipHeading"
  >
    <p>
       
      <em>
        Welcome to Complete news
      </em>
    </p>
  </div>
  <p>
     You have more than 500 articles in your palm 
  </p>
  <p>
    Login to store interesting articles to your dashboard
  </p>
  <p>
     To begin 
  </p>
  <p>
     Select a source from the source panel 
  </p>
  <p>
     ‚Üê 
  </p>
</div>
`;
